spec/integration/schema/or_spec.rb

```diff
@@ -18,6 +18,24 @@
     end
   end
 
+  context "with three predicates" do
+    subject(:schema) do
+      Dry::Schema.define do
+        required(:foo) { str? | int? | bool? }
+      end
+    end
+
+    it "returns success for valid input" do
+      expect(schema.(foo: "bar")).to be_success
+      expect(schema.(foo: 321)).to be_success
+      expect(schema.(foo: true)).to be_success
+    end
+
+    it "provides OR error message for invalid input where all sides failed" do
+      expect(schema.(foo: []).errors.to_h).to eql(foo: ["must be a string or must be an integer or must be boolean"])
```

Comment:
The `eql?` method fails without an explicit `to_h` compared with line 17 - https://github.com/dry-rb/dry-schema/pull/387/files#diff-678244486dbad600e6d79a5bb0cf3aa771877a010ddc921f082f7e3d3bc51e60R17